scripts/docgen/generate-docs.js

```diff
@@ -240,6 +243,63 @@ function fixAllLinks(htmlFiles) {
   return Promise.all(writePromises);
 }
 
+/**
+ * Generate an temporary abridged version of index.d.ts used to create
+ * Node docs.
+ */
+async function generateNodeSource() {
+  const sourceText = await fs.readFile(sourceFile, 'utf8');
+
+  // Parse index.d.ts. A dummy filename is required but it doesn't create a
+  // file.
+  let typescriptSourceFile = typescript.createSourceFile(
+    'temp.d.ts',
+    sourceText,
+    typescript.ScriptTarget.ES2015,
+    /*setParentNodes */ false
+  );
+
+  // Traverse AST to get blocks annotated with @webonly and store their
+  // start/end index.
+  const webOnlyBlocks = [];
+  function findWebOnlyBlocks(node) {
+    if (node.jsDoc) {
+      node.jsDoc.forEach(item => {
+        if (item.tags) {
+          item.tags.forEach(tag => {
+            if (tag.tagName.escapedText === 'webonly') {
+              webOnlyBlocks.push({ start: node.pos, end: node.end });
+            }
+          });
+        }
+      });
+    }
+    typescript.forEachChild(node, findWebOnlyBlocks);
```

Comment:
Can we early terminate when a node is deleted?

Reply:
Done